src/Aspire.Hosting.Azure/AzureEventHubsResource.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Azure.Provisioning.EventHubs;
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// Represents an Azure Event Hubs resource.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+/// <param name="configureConstruct">Callback to configure the Azure Event Hubs resource.</param>
+public class AzureEventHubsResource(string name, Action<ResourceModuleConstruct> configureConstruct)
+    : AzureConstructResource(name, configureConstruct), IResourceWithConnectionString
+{
+    internal List<(string Name, Action<IResourceBuilder<AzureEventHubsResource>, ResourceModuleConstruct, EventHub>? Configure)> Hubs { get; } = [];
+
+    /// <summary>
+    /// Gets the "serviceBusEndpoint" output reference from the bicep template for the Azure Storage resource.
+    /// </summary>
+    public BicepOutputReference ServiceBusEndpoint => new("serviceBusEndpoint", this);
+
+    /// <summary>
+    /// Gets the connection string template for the manifest for the Azure Event Hubs endpoint.
+    /// </summary>
+    public ReferenceExpression ConnectionStringExpression =>
+        ReferenceExpression.Create($"{ServiceBusEndpoint}");
```

Comment:
Why not use the `Endpoint=sb://...` variant? That way could append the entity path?

Reply:
I could -- but I'm duplicating what the Service Bus resource/component does. If we change this, we should change the ASB interfaces too. As you know, EH and ASB have the same lineage. It might be better to plan enhancements like this as a batch and update ASB and EH in the same PR. What do you think?